In [1]:
!pip install -U finance-datareader

In [2]:
import pandas as pd
import FinanceDataReader as fdr

In [3]:
fdr.__version__

'0.9.20'

In [5]:
fdr.StockListing??

In [7]:
# 한국거래소 상장종목 전체 가져오기
df_krx = fdr.StockListing("KRX")
df_krx.shape

(7110, 10)

In [8]:
df_krx.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7110 entries, 0 to 7109
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Symbol          7110 non-null   object        
 1   Market          7110 non-null   object        
 2   Name            7110 non-null   object        
 3   Sector          2422 non-null   object        
 4   Industry        2403 non-null   object        
 5   ListingDate     2422 non-null   datetime64[ns]
 6   SettleMonth     2422 non-null   object        
 7   Representative  2422 non-null   object        
 8   HomePage        2247 non-null   object        
 9   Region          2422 non-null   object        
dtypes: datetime64[ns](1), object(9)
memory usage: 611.0+ KB


### 생각보다 결측치가 많았다. 

In [9]:
df_krx.describe()

<ipython-input-9-8453f2ac474b>:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  df_krx.describe()


Symbol Market   Name         Sector Industry          ListingDate  \
count     7110   7110   7110           2422     2403                 2422   
unique    7110      3   7110            158     2251                 1784   
top     57FT30  KOSPI  서연탑메탈  특수 목적용 기계 제조업   기업인수합병  1976-06-30 00:00:00   
freq         1   5464      1            142       36                    8   
first      NaN    NaN    NaN            NaN      NaN  1956-03-03 00:00:00   
last       NaN    NaN    NaN            NaN      NaN  2021-02-05 00:00:00   

       SettleMonth Representative             HomePage Region  
count         2422           2422                 2247   2422  
unique           9           2307                 2226     22  
top            12월              -  http://www.dw.co.kr  서울특별시  
freq          2361              5                    2    977  
first          NaN            NaN                  NaN    NaN  
last           NaN            NaN                  NaN    NaN

1. symbol : count와 unique값이 같네 다행이도 
2. Market : 코스피, 코스닥, 코덱스로 구성되어 있고 KOSPI가 가장 많네 
3. Name : 2021년 1월에 비해 5000개가 상장을 더했네... 어마어마 하네 
4. Sector : 분야별로 데이터를 나누어놓은 것으로 특수목적용 기계제조업이 142개로 가장 많았다. 
5. Industry : 산업은 기업인수합병이 가장 많다. 
6. ListingDate : 1976-06-30에 8개의 기업이 장상을 가장 많이 했다고 나와 있다. 

In [10]:
# to_csv로 DataFrame을 데이터 저장용 파일인 csv 파일로 바꿀 수 있음 
df_krx.to_csv("krx.csv")

In [ ]:
pd.read_csv("krx.csv")